### PR TITLE
Attribute mapping added

### PR DESCRIPTION
**Pull request by Wulfheart**
Hello,
I wanted to map specific properties. For example a `name = "Something"` should be mapped to `public string Name`. This can be achieved like this:

```toml
[user]
name = "User1"
password = "PW"
``` 
```C#
using Tomlet;
public class Config
{
    [TomlProperty("user")]
    public User User { get; set; }
}

public class User
{
    [TomlProperty("name")]
    public string Name { get; set; }
    [TomlProperty("password")]
    public string Password { get; set; }
} 
```

Currently I made it available under https://www.nuget.org/packages/Wulfheart.Tomlet/2.1.0 until it gets eventually merged.

### CODE REVIEW

**Review by SamboyCoding — APPROVED** [review]

**Review by SamboyCoding — APPROVED** [review]

**Comment by SamboyCoding** [comment]
Looks good, the only question I would have is if you could possibly add some unit tests to validate this behaviour? Then I will gladly merge this.

**Comment by SamboyCoding** [comment]
Actually, ignore that, the test failed, you need to use `FirstOrDefault` instead of `First`

File: Tomlet/TomlSerializationMethods.cs [hunk 1]
```diff
@@ -187,8 +187,10 @@ internal static Deserialize<object> GetCompositeDeserializer(Type type)
                             var name = field.Name;
                             var m = System.Text.RegularExpressions.Regex.Match(field.Name, "<(.*)>k__BackingField");
                             if (m.Success)
-                                name = m.Groups[1].Value; 
+                                name = m.Groups[1].Value;
 
+                            var name1 = name;
+                            name = type.GetProperties().First(p => p.Name == name1).GetCustomAttribute<TomlPropertyAttribute>()?.GetMappedString() ?? name;
```
Comment (SamboyCoding) [inline Tomlet/TomlSerializationMethods.cs]: Please use FirstOrDefault here to not crash if the object being serialized is a field, not a property

**Review by SamboyCoding — CHANGES_REQUESTED** [review]

**Comment by Wulfheart** [comment]
Thanks for your comments. 

How can I run the tests locally with the CLI or Rider? Anywhere to look?

**Comment by SamboyCoding** [comment]
Rider should pick up the tests automatically, if you go to the Unit Tests tool window (View -> Tool Windows -> Unit Tests). Alternatively, the command is `dotnet test` from the solution folder

File: Tomlet/TomlSerializationMethods.cs (hunk 1 above)
Comment (Wulfheart) [reply Tomlet/TomlSerializationMethods.cs]: Changed

**Comment by Wulfheart** [comment]
@SamboyCoding Added tests and serialization also works now.

**Review by SamboyCoding — APPROVED** [review]

**Comment by SamboyCoding** [comment]
The build is failing because the resx file hasn't been recompiled, but I'll trust you if you say it works now, and the changes look like they make sense, I'll merge this in a bit and recompile the resx myself, then push to nuget.

**Comment by Wulfheart** [comment]
image

😉

Thank you! I'm looking forward to it.

**Comment by Wulfheart** [comment]
Thanks for merging. Is there already a release on Nuget?

**Comment by SamboyCoding** [comment]
Not yet, but I'm working on it now. Got one issue to fix then I'll publish and update you here

**Comment by SamboyCoding** [comment]
@Wulfheart The build is now validating on nuget - I anticipate it becoming available within 5-10 minutes.

**Pull request merged**